source/jormungandr/jormungandr/interfaces/v1/JourneyCommon.py

```diff
@@ -0,0 +1,264 @@
+        parser_get.add_argument("bike_speed", type=float_gt_0)
+        parser_get.add_argument("bss_speed", type=float_gt_0)
+        parser_get.add_argument("car_speed", type=float_gt_0)
+
+    def parse_args(self, region=None, uri=None):
+        args = self.parsers['get'].parse_args()
+
+        # We set default modes for fallback modes.
+        # The reason why we cannot put default values in parser_get.add_argument() is that, if we do so,
+        # fallback modes will always have a value, and traveler_type will never override fallback modes.
+        if args.get('origin_mode') is None:
+            args['origin_mode'] = ['walking']
+        if args.get('destination_mode') is None:
+            args['destination_mode'] = ['walking']
+
+        #we transform the origin/destination url to add information
+        if args['origin']:
+            args['origin'] = transform_id(args['origin'])
+        if args['destination']:
+            args['destination'] = transform_id(args['destination'])
+        if not args['datetime']:
+            args['datetime'] = args['_current_datetime']
+
+        args['original_datetime'] = args['datetime']
+
+        if args['data_freshness'] is None:
+            # retrocompatibilty handling
+            args['data_freshness'] = \
+                'adapted_schedule' if args['disruption_active'] is True else 'base_schedule'
+
+        if args.get('traveler_type'):
+            traveler_profile = TravelerProfile.make_traveler_profile(region, args['traveler_type'])
+            traveler_profile.override_params(args)
+
+        # for last and first section mode retrocompatibility
+        if 'first_section_mode' in args and args['first_section_mode']:
+            args['origin_mode'] = args['first_section_mode']
+        if 'last_section_mode' in args and args['last_section_mode']:
+            args['destination_mode'] = args['last_section_mode']
+
+        if args.get('max_duration_to_pt'):
+            # retrocompatibility: max_duration_to_pt override all individual value by mode
+            args['max_walking_duration_to_pt'] = args['max_duration_to_pt']
+            args['max_bike_duration_to_pt'] = args['max_duration_to_pt']
+            args['max_bss_duration_to_pt'] = args['max_duration_to_pt']
+            args['max_car_duration_to_pt'] = args['max_duration_to_pt']
+
+        # TODO : Changer le protobuff pour que ce soit propre
+        if args['destination_mode'] == 'vls':
+            args['destination_mode'] = 'bss'
+        if args['origin_mode'] == 'vls':
+            args['origin_mode'] = 'bss'
+
+        if region:
+            self.region = i_manager.get_region(region)
+            if uri:
+                objects = uri.split('/')
+                if objects and len(objects) % 2 == 0:
+                    args['origin'] = objects[-1]
+                else:
+                    abort(503, message="Unable to compute journeys "
+                                   "from this object")
```

Comment:
This block seems strange here. Maybe exposing it in another function and returning region instead of setting self.region?